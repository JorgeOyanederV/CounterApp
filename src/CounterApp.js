import React, { useState } from 'react';
import PropTypes from 'prop-types';

const CounterApp = ({ value = 0 }) => {

   const [estado, setEstado] = useState(value);

   return (
      <>
         <h1 class='flex-center'>CounterApp</h1>
         <h2 class='flex-center'> {estado} </h2>

         <button onClick={() => { setEstado(estado + 1) }} > Sumar 1</button>
         <button onClick={() => { setEstado(estado - 1) }} > Restar 1</button>
         <button onClick={() => { setEstado(value) }} > Resetear</button>
      </>
   );
}
CounterApp.propTypes = {
   value: PropTypes.number,
}


export default CounterApp;