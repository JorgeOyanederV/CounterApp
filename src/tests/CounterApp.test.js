import React, { Component } from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import CounterApp from '../CounterApp';

describe('Pruebas de <counterApp />', () => {
   test('Debe de mostrar el valor por defecto cuando no se le envia el valor', () => {
      const wrapper = shallow(<CounterApp />);
      expect(wrapper).toMatchSnapshot();
      const valorPorDefecto = wrapper.find('h2');
      expect(valorPorDefecto.text().trim()).toBe("0");
   })

})