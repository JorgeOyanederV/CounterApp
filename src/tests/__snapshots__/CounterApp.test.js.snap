// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de <counterApp /> Debe de mostrar el valor por defecto cuando no se le envia el valor 1`] = `ShallowWrapper {}`;
